### PR TITLE
liquidity: add custom swap amounts to autoloop

### PR DESCRIPTION
**Pull request by carlaKC**
This PR adds the ability for clients to configure custom min/max swap amounts for autolooper. Before this PR, the minimum swap size was pegged to the server's minimum value, which could result in multiple small swaps for a single channel. Adding a configurable minimum allows clients to set their own minimum, tradeoff here is that a larger minimum means that we drop further below our target incoming %, but we make larger swaps so there is potential fee saving. 

High level tracking task: #319

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: docs/autoloop.md [hunk 1]
```diff
@@ -191,6 +191,33 @@ The default value for this parameter is 24hours, and it can be updated as follow
...
 
+### Swap Size
+By default, the autolooper will execute a swap when the amount that needs to be
+rebalanced within a channel is equal to the swap server's minimum swap size. 
+This means that it will dispatch swaps more regularly, and ensure that channels 
+are not run down too far below their configured threshold. If you are willing 
+to allow your liquidity to drop further than the minimum swap amount below your 
+threshold, a custom minimum swap size can be set. If autolooper is configured 
+with a larger minimum swap size, it will allow channels to drop further below
+their target threshold, but will perform fewer swaps, potentially on saving
```
Comment (guggero) [inline docs/autoloop.md:202]: nit: s/on saving/saving on/

File: docs/autoloop.md [hunk 2]
```diff
@@ -191,6 +191,33 @@ The default value for this parameter is 24hours, and it can be updated as follow
...
+
+Swaps are also limited to the maximum swap amount advertised by the server. If
+you would like to reduce the size of swap that autoloop created, this value can 
+also be configured. 
+
+```
+loop setparams --maxamt={amount in satoshis}
+```
+
+The server's current terms are provided by the `terms` cli command. Note that
```
Comment (guggero) [inline docs/autoloop.md:217]: nit: specify `loop terms` instead of just `terms`.

**Review by guggero — APPROVED** [review]
Nice and clean set of changes, LGTM :100:

File: liquidity/liquidity.go [hunk 3]
```diff
@@ -347,6 +368,47 @@ func (p Parameters) validate(minConfs int32) error {
...
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validateRestrictions checks that client restrictions fall within the server's
+// restrictions.
+func validateRestrictions(server, client *Restrictions) error {
```
Comment (bhandras) [inline liquidity/liquidity.go:381]: nit: we could also return the clipped boundaries in one go here.

**Review by bhandras — APPROVED** [review]
LGTM, sorry for the late review. Nice change! :1st_place_medal:

**Pull request merged**